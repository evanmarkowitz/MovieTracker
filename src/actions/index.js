export const setMovies = (movies) => ({
  type: 'SET_MOVIES',
  movies
})

export const setLoggedInUser = (user) => ({
  type: 'SET_LOGGED_IN_USER',
  user
})

export const signOutUser = (user) => ({
  type: 'SIGN_OUT_USER',
  user
})
 
export const getFavorites = (favorites) => ({
  type: 'GET_FAVORITES',
  favorites
})

// export const addFavorites = (favorites) => ({
//   type: 'ADD_FAVORITES',
//   favorites
// })
