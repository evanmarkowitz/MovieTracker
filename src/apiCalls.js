export const fetchMovies = async () => {
  try {
    const response = await fetch('https://api.themoviedb.org/3/movie/now_playing?api_key=b6a9c244f525ef1a2fe0110d010ba4aa&language=en-US&include_adult=false&page=1')
    if(!response.ok){
      throw new Error('Error fetching movies')
    } 
    const movies = await response.json()
    return movies.results
  }
  catch(error){
    throw new Error(error.message)
  }
}

