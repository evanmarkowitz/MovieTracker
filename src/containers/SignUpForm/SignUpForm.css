.login-page {
  background-image: url('../../images/lego-background.jpg');
  background-size: cover;
  min-height: 100vh;
  display: flex;
  align-content: center;
  flex-direction: column;
}

h2 {
  align-self: center;
  margin: 20px;
  background: #333333;
  color: white;
  width: 350px;
}

