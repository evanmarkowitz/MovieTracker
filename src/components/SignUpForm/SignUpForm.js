import React, { Component } from 'react';
import { fetchNewUser } from '../../apiCalls';
import './SignUpForm.css';

class SignUpForm extends Component {
    constructor() {
        super();
        this.state = {
            name: '',
            email: '',
            password: '',
            error: ''
        }
    }

    handleChange = (e) => {
        this.setState({[e.target.name]: e.target.value})
    }

    handleSubmit = async (e) => {
        e.preventDefault();
        const newUser = {
            name: this.state.name,
            email: this.state.email,
            password: this.state.password
        }
        try {
          const postNewUser = await fetchNewUser('http://localhost:3000/api/users/new', newUser)
          this.setState({error: ''})
        } catch(error) {
          this.setState({error: error.message})
        }
        this.clearInputs()
    }
    clearInputs = () => {
      this.setState({
        name: '',
        email: '',
        password: ''
      })
    }

    render() {
        return (
            <>
                <h2>Create An Account :)</h2>
                <form>
                    <label htmlFor="name-input">Name:</label>
                    <input type="text" placeholder="Name"
                        name='name'
                        value={this.state.name}
                        onChange={this.handleChange} />
                    <label htmlFor="email-input">Email:</label>
                    <input type="text" placeholder="Email"
                        name='email'
                        value={this.state.email}
                        onChange={this.handleChange} />
                    <label htmlFor="password-input">Password:</label>
                    <input type="text" placeholder="Password"
                        name='password'
                        value={this.state.password}
                        onChange={this.handleChange} />
                    <button onClick={this.handleSubmit}>Submit</button>
                </form>
            {this.state.error && <h2>{this.state.error}</h2>}
            </>
        )
    }
}

export default SignUpForm;