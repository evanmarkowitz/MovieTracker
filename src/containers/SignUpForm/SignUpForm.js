import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchNewUser } from '../../apiCalls';
import { userLogin } from '../../apiCalls';
import { setLoggedInUser } from '../../actions';
import CustomForm from '../../components/Shared/CustomForm'
import './SignUpForm.css';

class SignUpForm extends Component {
    constructor() {
        super();
        this.state = {
            name: '',
            email: '',
            password: '',
            isLoggedIn: false,
            error: ''
        }
    }

    handleChange = (e) => {
        this.setState({ [e.target.name]: e.target.value })
    }

    handleSubmit = async (e) => {
        e.preventDefault();
        const newUser = {
            name: this.state.name,
            email: this.state.email,
            password: this.state.password
        }
        try {
            const postNewUser = await fetchNewUser('http://localhost:3000/api/users/new', newUser)
            let options = { email: this.state.email, password: this.state.password }
            let user = await userLogin('http://localhost:3000/api/users', options)
            this.props.setLoggedInUser(user.data)
            this.setState({ isLoggedIn: true })
            this.setState({ error: '' })
        } catch (error) {
            this.setState({ error: error.message })
        }
        this.clearInputs()
    }
    
    clearInputs = () => {
        this.setState({
            name: '',
            email: '',
            password: ''
        })
    }

    render() {
        return (
            <>
                <CustomForm 
                    title={'Create an Account :)'}
                    formFields={[
                        {
                            type: 'name',
                            value: this.state.name,
                            onChange: this.handleChange,
                        },
                        {
                            type: 'email',
                            value: this.state.email,
                            onChange: this.handleChange,
                        },
                        {
                            type: 'password',
                            value: this.state.password,
                            onChange: this.handleChange
                        }
                    ]}
                    onSubmit={this.handleSubmit}
                    isLoggedIn={this.state.isLoggedIn}
                    error={
                        {isError: this.state.error,
                        message: 'Error signing up'}
                    }   
                />
            </>
        )
    }
}

const mapDispatchToProps = dispatch => ({
    handleSubmit: () => dispatch(this.setState(this.state.isLoggedIn)),
    setLoggedInUser: (user) => dispatch(setLoggedInUser(user))
})
const mapStateToProps = (state) => ({
    isLoggedIn: state.isLoggedIn
})
export default connect(mapStateToProps, mapDispatchToProps)(SignUpForm)