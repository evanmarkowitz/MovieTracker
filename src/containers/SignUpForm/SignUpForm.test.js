import { SignUpForm, mapDispatchToProps, mapStateToProps } from './SignUpForm';
import { setLoggedInUser } from '../../actions';
import React, {Component} from 'react';
import { shallow } from 'enzyme';
import { fetchStoreProperties } from '../../apiCalls';

jest.mock("../../apiCalls", () => ({
  fetchStoreProperties: jest.fn().mockImplementation(()=> {
    return Promise.resolve({id: 1, name: "fake", emai: "fake", password: "fake"})
  })
 }))

describe('SignUpFormContainer', () => {
    describe('mapStateToProps', () => {
        it('should', () => {
            const state = {isLoggedIn: false};
            
            const mappedProps = mapStateToProps(state);

            expect(mappedProps).toEqual(state);
        })
    });

    describe('mapDispatchToProps', () => {
        it('should dispatch with a setLoggedInUser action when handleSubmit is called', () => {
            const mockDispatch = jest.fn();
            const mockUser = {name: 'Evan'};
            const mockAction = setLoggedInUser(mockUser);

            const mappedProps = mapDispatchToProps(mockDispatch);
            mappedProps.setLoggedInUser(mockUser);

            expect(mockDispatch).toHaveBeenCalledWith(mockAction);
        })
    })
    describe('Match Snapshot', () => {
        let wrapper;
        const isLoggedIn = false;
        beforeEach(() => {
            wrapper = shallow(<SignUpForm isLoggedIn={isLoggedIn}/>)
        })
        it('should match the snapshot', () => {
            expect(wrapper).toMatchSnapshot()
        })
    })
    describe('Update State Based On Inputs', () => {
      let wrapper
      beforeEach (() => {
        wrapper = shallow(<SignUpForm isLoggedIn={false}/>)
      })
      it(' should clear inputs', () => {
        wrapper.setState({name: 'e', email: 'e', password: 'e' })
        wrapper.instance().clearInputs()
        expect(wrapper.state()).toEqual({name: '', email: '', password: '', isLoggedIn: false, error: ''})
      })
      it('should handle change', () => {
        const mockEvent = { target: { name: 'name', value: 'Andy'} };
        const expected = 'Andy';
        wrapper.instance().handleChange(mockEvent);
        expect(wrapper.state('name')).toEqual(expected)
      })
    })
    describe('Handle Submit', () => {
      let wrapper;
      let mockEvent;
      
      beforeEach (() => {
        wrapper = shallow(<SignUpForm isLoggedIn={false}/>)
        mockEvent = jest.fn()
        window.fetch = jest.fn().mockImplementation(() => {
          return Promise.resolve({
            ok: true,
            json: () => Promise.resolve({title: 'e'})
          })
        })
        wrapper.instance().clearInputs = jest.fn();
      })
      it('should call fetchStoreProperties submit is called', () => {
        wrapper.instance().handleSubmit({preventDefault: jest.fn()});
        expect(fetchStoreProperties).toHaveBeenCalled();
      })
      it('should reset state when submit is called', () => {
        wrapper.instance().setState({error: 'test error', isLoggedIn: false})
        wrapper.instance().handleSubmit({preventDefault: jest.fn()});
        wrapper.state({name: '',
        email: '',
        password: '',
        isLoggedIn: true,
        error: ''})

      })
      
    })
}) 