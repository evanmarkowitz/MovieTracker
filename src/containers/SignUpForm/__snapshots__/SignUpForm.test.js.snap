// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpFormContainer Match Snapshot should match the snapshot 1`] = `ShallowWrapper {}`;
